### PR TITLE
ARACHNE-2121-fixed some issues on tenant page

### PR DESCRIPTION
**Pull request by agackovka**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java [hunk 1]
```diff
@@ -212,6 +217,16 @@ public DS updateInAnyTenant(DS dataSource)
         return savedDataSource;
     }
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Override
+    public DS updateDataSourceTenants(final DS dataSource)
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java]: Why do you put the method in community, if it is used only in enterprise?

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseStudyToStudyDTOConverter.java [hunk 2]
```diff
@@ -82,9 +83,17 @@ public DTO convert(final S source) {
...
         );
 
+        final Tenant studyTenant = source.getTenant();
+        
         for (final StudyDataSourceLink studyDataSourceLink : foundLinks) {
+            
             final DataSourceDTO dataSourceDTO = conversionService.convert(studyDataSourceLink,
                     DataSourceDTO.class);
+            
+            if (!studyDataSourceLink.getDataSource().getTenants().contains(studyTenant)) {
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseStudyToStudyDTOConverter.java]: What about performance and amount of queries with such approach?

Comment (agackovka) [reply src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseStudyToStudyDTOConverter.java]: Contains will be comparing ids, so entities will not be initialized

File: src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java (hunk 1 above)
Comment (agackovka) [reply src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java]: done

**Pull request merged**